src/js/components/Box/StyledBox.js

```diff
@@ -59,8 +59,8 @@ const basisStyle = css`
 // we assume we are in the context of a Box going the other direction
 // TODO: revisit this
 const directionStyle = css`
-  ${props => props.directionProp === 'row' && 'min-height: 0;'}
-  ${props => props.directionProp === 'column' && 'min-width: 0;'}
+  min-width: 0;
```

Comment:
`Adding both min-width: 0 and min-height: 0 always. This helps to avoid having to set a given direction for Box just to fix flexbugs
Fixed focus management for CheckBox to align with our other inputs`

Reply:
without it I'm facing overflow issues when using `<Text truncate={true} />`

Reply:
I seem to recall some situations where this isn't the right behavior. But, I'm having trouble remembering the details :(
This is the one change I am most nervous about causing an unexpected impact.

Reply:
I see. hum I cannot think of a way to fix that without creating a wrapper box, just to avoid this problem. I would much rather know for sure that we don't want to always use both.

Reply:
I've tested with with OneSphere and grommet-site. I'm not seeing any issues. So, I'm OK with this. Something might rear it's head later, but I don't want to hold this up for that.